mods/_fd/immersive_sounds/code/machinery.dm

```diff
@@ -0,0 +1,26 @@
+/obj/machinery
+	var/datum/sound_token/w_sound_token
+	var/w_sound_id
+	var/list/working_sounds
+	var/list/action_sound
+	var/startup_sound
+	var/shutdown_sound
+	var/working_volume
+
+/obj/machinery/proc/play_working_soundloop(rn = 10, flf = 1)
+	if(!working_sounds)
+		return
+	if(!w_sound_id)
+		w_sound_id = "[type]_[sequential_id(type)]"
+	if(!w_sound_token)
+		if(startup_sound)
+			playsound(src, startup_sound, (working_volume+5), TRUE)
+			sleep(length(startup_sound))
```

Comment:
Не стоит использовать `sleep`. Он буквально ПОЛНОСТЬЮ остановит ВСЮ(всю - в теории. Если брать пример тех же Суровых. По факту - только эту махину) машинерию. Замени на `spawn`

Reply:
что

Reply:
ок

Reply:
[sleep(Delay) | Pause the current proc (and its callers) for a specified amount of time. If no delay is specified, it will be scheduled to resume as soon as other immediately pending events are processed.](https://www.byond.com/docs/ref/#/proc/sleep)

Стопит ПОЛНОСТЬЮ машинерию до конца sleep(lenght(...))

Reply:
Спавн не работает из-за того, что идёт после него
mods\_fd\immersive_sounds\code\machinery.dm:19:error: : invalid expression

Reply:
Лодка из дерьма; Мы в любом случае не можем использовать СЛИП, ведь это в миг заставить хрякнуть 99% Машинерии, которой будет говорить: Дорогой. А ты сегодня СПИИИИШ

Reply:
Попробуй применить тогда addtimer.

Reply:
ok

Reply:
Мне кажется какая-то хуйня с указыванием времени для addtimer, поэтому он тупо не выполняется.

Reply:
а, нет, это просто из-за того, что я отдельно вынес в ещё один прок из-за ограничений addtimer

Reply:
сделано